malwarecage/web/src/components/QuickQuery.js

```diff
@@ -0,0 +1,129 @@
+import React, {Component} from 'react';
+
+import { ConfirmationModal } from "@malwarefront/ui";
+import api from "@malwarefront/api";
+
+import QuickQueryAddModal from "./QuickQueryAddModal";
+
+
+class QuickQuery extends Component {
+    state = {
+        queries: [],
+        quickQueryId: null,
+        isAddModalOpen: false,
+        isDeleteModalOpen: false,
+        modalError: ""
+    }
+
+    getQuickQueries = async () => {
+        try {
+            let response = await api.getQuickQueries(this.props.type)
+            this.setState({queries: response.data})
+        } catch (error) {
+            console.log(error)
+        }
+    }
+
+    componentDidMount() {
+        this.getQuickQueries()
+    }
+
+    submitQuickQuery = async (name) => {
+        if (!this.props.query || this.props.queryError !== false) {
+            this.setState({modalError: "You can't save empty or broken query. Search with it first."})
```

Comment:
I don't really like that message. We should just disable "Add +" button instead of handling this inside modal.

Reply:
Fixed